improver/utilities/spatial.py

```diff
@@ -230,8 +234,20 @@ def create_difference_cube(
             Cube containing the differences calculated along the
             specified axis.
         """
-        points = cube.coord(coord_name).points
+        axis = cube.coord(coord_name)
+        points = axis.points
         mean_points = (points[1:] + points[:-1]) / 2
+        if self._axis_wraps_around_meridian(axis, cube):
+            if type(axis.coord_system) != GeogCS:
+                raise ValueError(
+                    "DifferenceBetweenAdjacentGridSquares does not currently support cubes with "
+                    "circular x-axis that do not use a geographic (i.e. latlon) coordinate system."
+                )
+            max_value = 360
+            extra_mean_point = (
+                np.mean([points[-1], (points[0] + max_value)]) % max_value
+            )
+            mean_points = np.hstack([mean_points, extra_mean_point])
```

Comment:
...Thinking about it, for the symmetrical case above, the np.mean(...) line always resolves to 180, so I could get rid of it altogether and just always put the wrap around point at 180.

The above does better handle the case where the cube uses 0 -> 360 instead of -180 -> 180 but maybe this never comes up.

I suppose the line could also be used as a test. I.e. if np.mean(...) != 180 raise ValueError("Difference Plugin only handles circular cubes if their points are spread symmetrically about a plane intersecting the prime meridian").

Reply:
There's a nice function I knew existed somewhere for calculating the circular mean. It's in scipy https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.circmean.html . This might be helpful but it does still need to be told the bounds to know where to wrap around so I don't know if it solves every problem but should simplify the offset case.

Id keep this function as general as possible. so I wouldn't raise an error, at worst a warning.